nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentManager.java

```diff
@@ -32,11 +32,40 @@
  */
 public interface CommentManager {
 
+    /**
+     * Gets comments of a document.
+     *
+     * @param docModel the document model
+     * @return the list of comments
+     */
     List<DocumentModel> getComments(DocumentModel docModel);
 
+    /**
+     * Gets comments of a document.
+     *
+     * @param session the core session
+     * @param docModel the document model
+     * @return the list of comments
+     * @since 10.3
+     */
+    List<DocumentModel> getComments(CoreSession session, DocumentModel docModel);
```

Comment:
what's the point specifying a session ?

Reply:
You have to specify it in the case of the deletion of the commented document, since you cannot use its associated session.